src/main/java/org/sagebionetworks/web/server/servlet/SynapseClientImpl.java

```diff
@@ -1609,24 +1538,72 @@ public Long getOpenRequestCount(String currentUserId, String teamId)
 			// now go through and create a MembershipInvitationBundle for each
 			// pair
 
-			List<Long> userIds = new ArrayList<>();
+			List<MembershipInvtnSubmission> invitationsToUsers = new ArrayList<>();
+			List<MembershipInvtnSubmission> invitationsToEmails = new ArrayList<>();
+			// Sort the results into the two types
 			for (MembershipInvtnSubmission invite : invitations.getResults()) {
+				String inviteeId = invite.getInviteeId();
+				String inviteeEmail = invite.getInviteeEmail();
+				if (inviteeId != null) {
+					invitationsToUsers.add(invite);
+				} else if (inviteeEmail != null) {
+					invitationsToEmails.add(invite);
+				}
+			}
+
+			// Get profiles of existing user invitees
+			List<Long> userIds = new ArrayList<>();
+			for (MembershipInvtnSubmission invite : invitationsToUsers) {
 				userIds.add(Long.parseLong(invite.getInviteeId()));
 			}
 			Map<String, UserProfile> userId2UserProfile = getUserProfiles(userIds, synapseClient);
-			
-			for (MembershipInvtnSubmission invite : invitations.getResults()) {
+			// Add invitations to return list
+			for (MembershipInvtnSubmission invite : invitationsToUsers) {
 				UserProfile profile = userId2UserProfile.get(invite.getInviteeId());
 				OpenTeamInvitationBundle b = new OpenTeamInvitationBundle(invite, profile);
 				returnList.add(b);
 			}
+			for (MembershipInvtnSubmission invite : invitationsToEmails) {
+				// Invitations to new users have a null profile
+				returnList.add(new OpenTeamInvitationBundle(invite, null));
+			}
 
 			return returnList;
 		} catch (SynapseException e) {
 			throw ExceptionUtil.convertSynapseException(e);
 		} 
 	}
-	
+
+	@Override
+	public MembershipInvtnSubmission getMembershipInvitation(MembershipInvtnSignedToken token) throws RestServiceException {
```

Comment:
Simple new methods like this should be done in the SynapseJavascriptClient instead of introducing new GWT RPC calls.